### DISCUSSION TITLE
Change Reactive -> MVUX

### DISCUSSION THREAD

**Discussion by dansiegel** [unoplatform/uno.extensions#1119] [Ideas]
Uno.Extensions.Reactive is not in any shape way or form Reactive. It does not use IObservable, or have a dependency on System.Reactive. While some concepts may be similar, this naming only leads to confusion in the community when people who understand Reactive programming see this and do not get the programming model they are expecting based on the name. Reactive is a well established pattern in the .NET community but this pattern revolves around IObservable and System.Reactive.

As a result the package name needs to be changed to MVUX as what the Reactive package actually is, is an implementation of MVU for Xaml bindings. This keeps user expectations in line with what the package actually delivers.

The package / namespace should be renamed `Uno.Extensions.Reactive` -> `Uno.Extensions.Mvux`

**Comment by francoistanguay** [comment]
According to https://www.techtarget.com/searchapparchitecture/definition/reactive-programming#:~:text=Reactive%20programming%20describes%20a%20design,a%20user%20makes%20an%20inquiry.

"Reactive programming describes a design paradigm that relies on asynchronous programming logic to handle real-time updates to otherwise static content. It provides an efficient means -- the use of automated data streams -- to handle data updates to content whenever a user makes an inquiry."

Reactive programming is a pattern, not something unique to .NET, RX or RXUI.

Chaning naming from Reactive to MVUXC would be a breaking change that wouldnt provide much value and would rather focus on better documentation and examples to clarify our intent.

**Reply by dansiegel** [reply]
When developers discuss Reactive programming in .NET there is a very specific pattern they are discussing. Uno.Extensions.Reactive is not it. It also was not built to be "Reactive" it was built to provide support for MVU which further confuses conversations around what is the package and when/why should you use it. While renaming the package would indeed be breaking it is a needed break. Uno should support the reactive programming model correctly and discuss this package for what it is an MVU support library.

**Reply by weitzhandler** [reply]
I don't know what's the adoption volume of MVUX, but I believe it's still in its early stages where such a rename is affordable.  
My personal vote for reuniting the terminology and rename the namespace to `Mvux`. Potentially it can also avoid confusion of newcomers reading the docs and looking at the source code.
